exp/bufferpool/pool.go

```diff
@@ -4,49 +4,161 @@ package bufferpool
 import "sync"
 
 const (
-	nBuckets = 20
+	defaultMinSize     = 1024
+	defaultBucketCount = 10
 )
 
-// A default global pool.
-var Default Pool
-
-// A pool of buffers, in variable sizes.
-type Pool struct {
-	buckets [nBuckets]sync.Pool
-}
-
-// Get a buffer of the given length. Its capacity may be larger than the
-// requested size.
-func (p *Pool) Get(size int) []byte {
-	for i := 0; i < nBuckets; i++ {
-		capacity := 1 << i
-		if capacity >= size {
-			var ret []byte
-			item := p.buckets[i].Get()
-			if item == nil {
-				ret = make([]byte, capacity)
-			} else {
-				ret = item.([]byte)
-			}
-			ret = ret[:size]
-			return ret
-		}
+var pool Pool = &BasicPool{}
```

Comment:
Is there a reason for this to no longer be public?

I might name it `defaultPool`; `pool` is a bit too generic. Or keep it as `Default` if it's going to be exported.

Reply:
Two reasons:

1. avoid it being set to `nil`
2. force calls to go through the top-level `Get` and `Put` methods, since they handle things like zeroing memory, which keeps the interface implementation simple.

Reply:
>I might name it defaultPool; pool is a bit too generic. 

Done.